### PR TITLE
Rule class

### PR DESCRIPTION
**Pull request by dorukozturk**
*Issue #, if available:*

*Description of changes:*
1) Implemented a Rule class for easier extension of rules
2) Created a consolidated view of rules
3) Filtered out false positives with cluster roles
4) Added an option to export json

Screen Shot 2023-03-16 at 10 32 10 AM

### CODE REVIEW

File: hardeneks/__init__.py [hunk 1]
```diff
@@ -159,24 +219,26 @@ def run_hardeneks(
...
         resources = NamespacedResources(region, context, cluster, ns)
         resources.set_resources()
-        harden(resources, rules, "namespace_based")
-        console.print()
+        namespace_based_results = harden(resources, rules, "namespace_based")
+        results = results + namespace_based_results
+
+    print_consolidated_results(results)
 
     if export_txt:
```
Comment (joshkurz) [inline hardeneks/__init__.py:239]: Do we want users to be able to add all three export options at once? Or do we only want to allow one at a time?

File: hardeneks/cluster_wide/cluster_autoscaling/cluster_autoscaler.py [hunk 2]
```diff
@@ -39,184 +37,216 @@ def _get_policy_documents_for_role(role_name, iam_client):
     return actions
 
 
-def check_any_cluster_autoscaler_exists(resources: Resources):
+class check_any_cluster_autoscaler_exists(Rule):
+    _type = "cluster_wide"
```
Comment (joshkurz) [inline hardeneks/cluster_wide/cluster_autoscaling/cluster_autoscaler.py:41]: So I really like this method of setting the metadata. I'm wondering, should we take it further and put this into the config? each test we run in the config would have its own set of metadata? Possibly could allow for extensions that users could write on their own. 

Just a thought maybe another iteration we do something like that.

**Review by joshkurz — APPROVED** [review]
Really excellent PR. Nice Job!

File: hardeneks/__init__.py (hunk 1 above)
Comment (dorukozturk) [reply hardeneks/__init__.py:239]: I think it is ok for them to do all three export at once? I did it that way assuming if someone needs to see different outputs but run the tool only once

File: hardeneks/cluster_wide/cluster_autoscaling/cluster_autoscaler.py (hunk 2 above)
Comment (dorukozturk) [reply hardeneks/cluster_wide/cluster_autoscaling/cluster_autoscaler.py:41]: definitely. That is where I was going :) This is to make the extension easier.

**Pull request merged**